koi_api/orm/sample.py

```diff
@@ -56,7 +59,25 @@ class ORMSample(db.Model):
         "ORMLabelRequest", back_populates="sample", lazy="dynamic"
     )
 
-    tags = db.relationship("ORMSampleTag", secondary=association_table)
+    tags = db.relationship("ORMAssociationTags", back_populates="sample", lazy="dynamic")
+
+    def purge_for_merge(self):
+        # collect all labels that will be lost ofter merging
+        labels_to_pruge = self.label.filter_by(mergeable=False).all()
+        for label in labels_to_pruge:
+            db.session.delete(label)
+
+        # collect all associations that will be lost after merging
+        tags_assoc_to_prune = self.tags.filter_by(mergeable=False).all()
+        for tag in tags_assoc_to_prune:
+            db.session.delete(tag)
+        db.session.commit()

```

Comment:
seems strange to commit here. Do we know that the merging is able to complete at this point?
Also: What happens when the application crashes at this point. The next steps will never be executed and the DB will contain an unwanted state, doesn't it?